<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Violin plot â€“ A Biomedical Visualization Atlas</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../About.html" rel="next">
<link href="../Distribution/Histogram.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-a113ddee828ef8ca18a4f67482cfc6d7.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../Distribution/Histogram.html">Distribution</a></li><li class="breadcrumb-item"><a href="../Distribution/ViolinPlot.html">Violin Plot</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/Bizard_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/DrRobinLuo/Bizard" title="Bizard GitHub website" class="quarto-navigation-tool px-1" aria-label="Bizard GitHub website"><i class="bi bi-github"></i></a>
    <a href="https://x.com/PengL_Robin" title="Peng Lab Twitter" class="quarto-navigation-tool px-1" aria-label="Peng Lab Twitter"><i class="bi bi-twitter"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-translate"></i></a>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Graph-Gallery.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Graph Gallery</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Distribution</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/Histogram.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Histogram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/ViolinPlot.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Violin Plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../About.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About Us</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#example" id="toc-example" class="nav-link active" data-scroll-target="#example">Example</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#data-preparation" id="toc-data-preparation" class="nav-link" data-scroll-target="#data-preparation">Data Preparation</a></li>
  <li><a href="#visualization" id="toc-visualization" class="nav-link" data-scroll-target="#visualization">Visualization</a>
  <ul class="collapse">
  <li><a href="#basic-violin-plot" id="toc-basic-violin-plot" class="nav-link" data-scroll-target="#basic-violin-plot">1. Basic Violin Plot</a></li>
  <li><a href="#horizontal-violin-plot" id="toc-horizontal-violin-plot" class="nav-link" data-scroll-target="#horizontal-violin-plot">2. Horizontal violin plot</a></li>
  <li><a href="#violin-plot-with-boxplot" id="toc-violin-plot-with-boxplot" class="nav-link" data-scroll-target="#violin-plot-with-boxplot">3. Violin Plot with Boxplot</a></li>
  <li><a href="#grouped-violin-plot" id="toc-grouped-violin-plot" class="nav-link" data-scroll-target="#grouped-violin-plot">4. Grouped Violin Plot</a></li>
  <li><a href="#half-violin-plot" id="toc-half-violin-plot" class="nav-link" data-scroll-target="#half-violin-plot">5. Half-Violin Plot</a></li>
  <li><a href="#violin-plot-using-the-ggstatsplot-package" id="toc-violin-plot-using-the-ggstatsplot-package" class="nav-link" data-scroll-target="#violin-plot-using-the-ggstatsplot-package">6. Violin Plot using the <code>ggstatsplot</code> Package</a></li>
  </ul></li>
  <li><a href="#application" id="toc-application" class="nav-link" data-scroll-target="#application">Application</a>
  <ul class="collapse">
  <li><a href="#basic-violin-plot-1" id="toc-basic-violin-plot-1" class="nav-link" data-scroll-target="#basic-violin-plot-1">1. Basic Violin Plot</a></li>
  <li><a href="#grouped-violin-plot-1" id="toc-grouped-violin-plot-1" class="nav-link" data-scroll-target="#grouped-violin-plot-1">2. Grouped Violin Plot</a></li>
  <li><a href="#half-violin-plot-1" id="toc-half-violin-plot-1" class="nav-link" data-scroll-target="#half-violin-plot-1">3. Half-Violin Plot</a></li>
  <li><a href="#violin-plot-with-boxplot-1" id="toc-violin-plot-with-boxplot-1" class="nav-link" data-scroll-target="#violin-plot-with-boxplot-1">4. Violin Plot with Boxplot</a></li>
  </ul></li>
  <li><a href="#reference" id="toc-reference" class="nav-link" data-scroll-target="#reference">Reference</a></li>
  <li><a href="#contributors" id="toc-contributors" class="nav-link" data-scroll-target="#contributors">Contributors</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/DrRobinLuo/Bizard/blob/main/Distribution/ViolinPlot.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/DrRobinLuo/Bizard/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../Distribution/Histogram.html">Distribution</a></li><li class="breadcrumb-item"><a href="../Distribution/ViolinPlot.html">Violin Plot</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Violin plot</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>A violin plot combines elements of a density plot and a box plot to visualize data distribution. It displays key statistical information, including the median, quartiles, minimum, and maximum values. Violin plots are particularly useful for comparing distributions across different groups, offering a more intuitive representation than traditional box plots by revealing the shape of the data distribution.</p>
<ul>
<li>Box Component: The white dot/line represents the median, while the black bar indicates the interquartile range (IQR).</li>
<li>Kernel Density Component: Shows the density of data points at different positions, providing insights into clustering patterns.</li>
</ul>
<section id="example" class="level2">
<h2 class="anchored" data-anchor-id="example">Example</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/ViolinPlot_demo.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%" alt="ViolinPlot DEMO"></p>
</figure>
</div>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<ul>
<li><p>System Requirements: Cross-platform (Linux/MacOS/Windows)</p></li>
<li><p>Programming language: R</p></li>
<li><p>Dependent packages: <code>tidyr</code>; <code>ggplot2</code>; <code>dplyr</code>; <code>gghalves</code>; <code>forcats</code>; <code>hrbrthemes</code>; <code>viridis</code>; <code>ggstatsplot</code>; <code>palmerpenguins</code></p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Installing necessary packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"tidyr"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"gghalves"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"gghalves"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading gghalves from CRAN ...            OK [230.4 Kb in 0.5s]
Successfully downloaded 1 package in 1.5 seconds.

The following package(s) will be installed:
- gghalves [0.1.4]
These packages will be installed into "~/work/Bizard/Bizard/renv/library/linux-ubuntu-noble/R-4.4/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing gghalves ...                       OK [installed binary and cached in 0.69s]
Successfully installed 1 package in 0.72 seconds.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"dplyr"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"ggplot2"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"forcats"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"forcats"</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"hrbrthemes"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"hrbrthemes"</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading hrbrthemes from CRAN ...          OK [846.9 Kb in 0.23s]
- Downloading extrafont from CRAN ...           OK [53.6 Kb in 0.29s]
- Downloading extrafontdb from CRAN ...         OK [6.8 Kb in 0.26s]
- Downloading Rttf2pt1 from CRAN ...            OK [267.4 Kb in 0.25s]
- Downloading gdtools from CRAN ...             OK [203.8 Kb in 0.23s]
- Downloading fontquiver from CRAN ...          OK [2.2 Mb in 0.25s]
- Downloading fontBitstreamVera from CRAN ...   OK [683.1 Kb in 0.27s]
- Downloading fontLiberation from CRAN ...      OK [4.3 Mb in 0.33s]
Successfully downloaded 8 packages in 3.3 seconds.

The following package(s) will be installed:
- extrafont         [0.19]
- extrafontdb       [1.0]
- fontBitstreamVera [0.1.1]
- fontLiberation    [0.1.0]
- fontquiver        [0.2.1]
- gdtools           [0.4.1]
- hrbrthemes        [0.8.7]
- Rttf2pt1          [1.3.12]
These packages will be installed into "~/work/Bizard/Bizard/renv/library/linux-ubuntu-noble/R-4.4/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing extrafontdb ...                    OK [installed binary and cached in 0.21s]
- Installing Rttf2pt1 ...                       OK [installed binary and cached in 0.21s]
- Installing extrafont ...                      OK [installed binary and cached in 0.21s]
- Installing fontBitstreamVera ...              OK [installed binary and cached in 0.22s]
- Installing fontLiberation ...                 OK [installed binary and cached in 0.3s]
- Installing fontquiver ...                     OK [installed binary and cached in 0.25s]
- Installing gdtools ...                        OK [installed binary and cached in 0.41s]
- Installing hrbrthemes ...                     OK [installed binary and cached in 0.78s]
Successfully installed 8 packages in 2.9 seconds.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"viridis"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"viridis"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"ggstatsplot"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"ggstatsplot"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading ggstatsplot from CRAN ...         OK [3.2 Mb in 0.64s]
- Downloading correlation from CRAN ...         OK [1.5 Mb in 0.36s]
- Downloading bayestestR from CRAN ...          OK [1 Mb in 0.33s]
- Downloading insight from CRAN ...             OK [2.2 Mb in 0.19s]
- Downloading datawizard from CRAN ...          OK [1.1 Mb in 0.25s]
- Downloading parameters from CRAN ...          OK [2.1 Mb in 0.18s]
- Downloading ggcorrplot from CRAN ...          OK [28.3 Kb in 0.35s]
- Downloading reshape2 from CRAN ...            OK [120.2 Kb in 0.2s]
- Downloading ggrepel from CRAN ...             OK [283.3 Kb in 0.17s]
- Downloading ggside from CRAN ...              OK [2.6 Mb in 0.27s]
- Downloading paletteer from CRAN ...           OK [425.9 Kb in 0.26s]
- Downloading prismatic from CRAN ...           OK [724 Kb in 0.3s]
- Downloading patchwork from CRAN ...           OK [3.2 Mb in 0.33s]
- Downloading performance from CRAN ...         OK [3 Mb in 0.27s]
- Downloading statsExpressions from CRAN ...    OK [3.2 Mb in 0.37s]
- Downloading afex from CRAN ...                OK [2.6 Mb in 0.27s]
- Downloading lmerTest from CRAN ...            OK [526 Kb in 0.22s]
- Downloading BayesFactor from CRAN ...         OK [6.8 Mb in 0.41s]
- Downloading coda from CRAN ...                OK [320.9 Kb in 0.16s]
- Downloading pbapply from CRAN ...             OK [98 Kb in 0.15s]
- Downloading hypergeo from CRAN ...            OK [346.5 Kb in 0.25s]
- Downloading elliptic from CRAN ...            OK [1.2 Mb in 0.19s]
- Downloading contfrac from CRAN ...            OK [23.9 Kb in 0.21s]
- Downloading deSolve from CRAN ...             OK [2.2 Mb in 0.17s]
- Downloading effectsize from CRAN ...          OK [792.2 Kb in 0.27s]
- Downloading PMCMRplus from CRAN ...           OK [1.2 Mb in 0.28s]
- Downloading gmp from CRAN ...                 OK [323.4 Kb in 0.16s]
- Downloading Rmpfr from CRAN ...               OK [1.2 Mb in 0.16s]
- Downloading SuppDists from CRAN ...           OK [232.7 Kb in 0.16s]
- Downloading kSamples from CRAN ...            OK [246.1 Kb in 0.33s]
- Downloading BWStest from CRAN ...             OK [147.9 Kb in 0.26s]
- Downloading rstantools from CRAN ...          OK [166.2 Kb in 0.16s]
- Downloading desc from CRAN ...                OK [326.9 Kb in 0.15s]
- Downloading RcppParallel from CRAN ...        OK [1.9 Mb in 0.18s]
- Downloading WRS2 from CRAN ...                OK [945.6 Kb in 0.28s]
- Downloading reshape from CRAN ...             OK [165.6 Kb in 0.27s]
- Downloading zeallot from CRAN ...             OK [57.7 Kb in 0.18s]
Successfully downloaded 37 packages in 14 seconds.

The following package(s) will be installed:
- afex             [1.4-1]
- BayesFactor      [0.9.12-4.7]
- bayestestR       [0.15.2]
- BWStest          [0.2.3]
- coda             [0.19-4.1]
- contfrac         [1.1-12]
- correlation      [0.8.7]
- datawizard       [1.0.1]
- desc             [1.4.3]
- deSolve          [1.40]
- effectsize       [1.0.0]
- elliptic         [1.4-0]
- ggcorrplot       [0.1.4.1]
- ggrepel          [0.9.6]
- ggside           [0.3.1]
- ggstatsplot      [0.13.0]
- gmp              [0.7-5]
- hypergeo         [1.2-13]
- insight          [1.1.0]
- kSamples         [1.2-10]
- lmerTest         [3.1-3]
- paletteer        [1.6.0]
- parameters       [0.24.2]
- patchwork        [1.3.0]
- pbapply          [1.7-2]
- performance      [0.13.0]
- PMCMRplus        [1.9.12]
- prismatic        [1.1.2]
- RcppParallel     [5.1.10]
- reshape          [0.8.9]
- reshape2         [1.4.4]
- Rmpfr            [1.0-0]
- rstantools       [2.4.0]
- statsExpressions [1.6.2]
- SuppDists        [1.1-9.8]
- WRS2             [1.1-6]
- zeallot          [0.1.0]
These packages will be installed into "~/work/Bizard/Bizard/renv/library/linux-ubuntu-noble/R-4.4/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing insight ...                        OK [installed binary and cached in 0.28s]
- Installing datawizard ...                     OK [installed binary and cached in 0.27s]
- Installing bayestestR ...                     OK [installed binary and cached in 0.28s]
- Installing parameters ...                     OK [installed binary and cached in 0.33s]
- Installing correlation ...                    OK [installed binary and cached in 0.3s]
- Installing reshape2 ...                       OK [installed binary and cached in 0.41s]
- Installing ggcorrplot ...                     OK [installed binary and cached in 0.67s]
- Installing ggrepel ...                        OK [installed binary and cached in 0.73s]
- Installing ggside ...                         OK [installed binary and cached in 0.72s]
- Installing prismatic ...                      OK [installed binary and cached in 0.23s]
- Installing paletteer ...                      OK [installed binary and cached in 0.47s]
- Installing patchwork ...                      OK [installed binary and cached in 0.78s]
- Installing performance ...                    OK [installed binary and cached in 0.31s]
- Installing lmerTest ...                       OK [installed binary and cached in 1.5s]
- Installing afex ...                           OK [installed binary and cached in 1.6s]
- Installing coda ...                           OK [installed binary and cached in 0.24s]
- Installing pbapply ...                        OK [installed binary and cached in 0.22s]
- Installing elliptic ...                       OK [installed binary and cached in 0.24s]
- Installing contfrac ...                       OK [installed binary and cached in 0.2s]
- Installing deSolve ...                        OK [installed binary and cached in 0.27s]
- Installing hypergeo ...                       OK [installed binary and cached in 0.24s]
- Installing BayesFactor ...                    OK [installed binary and cached in 1.5s]
- Installing effectsize ...                     OK [installed binary and cached in 0.32s]
- Installing gmp ...                            OK [installed binary and cached in 0.25s]
- Installing Rmpfr ...                          OK [installed binary and cached in 0.49s]
- Installing SuppDists ...                      OK [installed binary and cached in 0.21s]
- Installing kSamples ...                       OK [installed binary and cached in 0.22s]
- Installing BWStest ...                        OK [installed binary and cached in 0.37s]
- Installing PMCMRplus ...                      OK [installed binary and cached in 0.74s]
- Installing desc ...                           OK [installed binary and cached in 0.22s]
- Installing RcppParallel ...                   OK [installed binary and cached in 0.3s]
- Installing rstantools ...                     OK [installed binary and cached in 0.22s]
- Installing reshape ...                        OK [installed binary and cached in 0.29s]
- Installing WRS2 ...                           OK [installed binary and cached in 0.31s]
- Installing zeallot ...                        OK [installed binary and cached in 0.21s]
- Installing statsExpressions ...               OK [installed binary and cached in 0.75s]
- Installing ggstatsplot ...                    OK [installed binary and cached in 0.89s]
Successfully installed 37 packages in 19 seconds.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"palmerpenguins"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"palmerpenguins"</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading palmerpenguins from CRAN ...      OK [2.9 Mb in 0.19s]
Successfully downloaded 1 package in 0.3 seconds.

The following package(s) will be installed:
- palmerpenguins [0.1.1]
These packages will be installed into "~/work/Bizard/Bizard/renv/library/linux-ubuntu-noble/R-4.4/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing palmerpenguins ...                 OK [installed binary and cached in 0.25s]
Successfully installed 1 package in 0.3 seconds.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gghalves)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forcats)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hrbrthemes)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggstatsplot)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(palmerpenguins)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-preparation" class="level2">
<h2 class="anchored" data-anchor-id="data-preparation">Data Preparation</h2>
<p>We used the built-in R datasets (<code>iris</code>, <code>penguins</code>) and the <code>TCGA-BRCA.htseq_counts.tsv</code> dataset from <a href="https://xenabrowser.net/datapages/?hub=https://gdcv18.xenahubs.net:443">UCSC Xena DATASETS</a>. Selected genes were chosen for demonstration purposes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the TCGA-BRCA gene expression dataset from a processed CSV file  </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>data_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://raw.githubusercontent.com/DrRobinLuo/Bizard/refs/heads/main/files/TCGA-BRCA.htseq_counts_processed.csv"</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load built-in R dataset iris</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>data_wide <span class="ot">&lt;-</span> iris[ , <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>] <span class="co"># Take the data in columns 1-4 of the iris database as an example</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Load built-in R dataset penguins</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"penguins"</span>, <span class="at">package =</span> <span class="st">"palmerpenguins"</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>data_penguins <span class="ot">&lt;-</span> <span class="fu">drop_na</span>(penguins) <span class="co"># Remove missing values</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Manually create a demonstration dataset with grouped values </span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">name=</span><span class="fu">c</span>( <span class="fu">rep</span>(<span class="st">"A"</span>,<span class="dv">500</span>), <span class="fu">rep</span>(<span class="st">"B"</span>,<span class="dv">500</span>), <span class="fu">rep</span>(<span class="st">"B"</span>,<span class="dv">500</span>), <span class="fu">rep</span>(<span class="st">"C"</span>,<span class="dv">20</span>), <span class="fu">rep</span>(<span class="st">'D'</span>, <span class="dv">100</span>)  ),</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">value=</span><span class="fu">c</span>( <span class="fu">rnorm</span>(<span class="dv">500</span>, <span class="dv">10</span>, <span class="dv">5</span>), <span class="fu">rnorm</span>(<span class="dv">500</span>, <span class="dv">13</span>, <span class="dv">1</span>), <span class="fu">rnorm</span>(<span class="dv">500</span>, <span class="dv">18</span>, <span class="dv">1</span>), <span class="fu">rnorm</span>(<span class="dv">20</span>, <span class="dv">25</span>, <span class="dv">4</span>), <span class="fu">rnorm</span>(<span class="dv">100</span>, <span class="dv">12</span>, <span class="dv">1</span>) )</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>sample_size <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> </span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">num=</span><span class="fu">n</span>()) <span class="co"># Compute the sample size for each group  </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualization" class="level2">
<h2 class="anchored" data-anchor-id="visualization">Visualization</h2>
<section id="basic-violin-plot" class="level3">
<h3 class="anchored" data-anchor-id="basic-violin-plot">1. Basic Violin Plot</h3>
<p>Example 1: Basic Violin Plot with Manually Created Data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic Violin Plot</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>value, <span class="at">fill=</span>name)) <span class="sc">+</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-1.1BasicViolin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-1.1BasicViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-1.1BasicViolin-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-1.1BasicViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Basic Violin Plot
</figcaption>
</figure>
</div>
</div>
</div>
<p>Example 2: Basic Violin Plot with the iris Dataset</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform the `iris` dataset from wide format to long format. Use the gather function to collect the data in each column into two new columns named "MesureType" and "Val", so that one row represents one observation.</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>data_long_iris <span class="ot">&lt;-</span> data_wide <span class="sc">%&gt;%</span>  </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"MeasureType"</span>, <span class="at">value =</span> <span class="st">"Value"</span>)  </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_long_iris, <span class="fu">aes</span>(<span class="at">x =</span> MeasureType, <span class="at">y =</span> Value, <span class="at">fill =</span> MeasureType)) <span class="sc">+</span>  </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-1.2BasicViolin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-1.2BasicViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-1.2BasicViolin-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-1.2BasicViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Basic Violin Plot using the <code>iris</code> dataset
</figcaption>
</figure>
</div>
</div>
</div>
<p>Example 3: Violin Plot with TCGA-BRCA Gene Expression Data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>example_counts1 <span class="ot">&lt;-</span> data_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,] <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"sample"</span>,<span class="at">value =</span> <span class="st">"gene_expression"</span>,<span class="dv">3</span><span class="sc">:</span><span class="dv">1219</span>) <span class="co"># Select five example genes for visualization: A1BG, A1BG-AS1, A1CF, A2M, and A2M-AS1.</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(example_counts1, <span class="fu">aes</span>(<span class="at">x=</span>gene_name, <span class="at">y=</span>gene_expression, <span class="at">fill=</span>gene_name)) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-1.3BasicViolin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-1.3BasicViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-1.3BasicViolin-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-1.3BasicViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Basic Violin Plot using the <code>TCGA-BRCA</code> dataset
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="horizontal-violin-plot" class="level3">
<h3 class="anchored" data-anchor-id="horizontal-violin-plot">2. Horizontal violin plot</h3>
<p>The x and y axes can be flipped using <code>coord_flip()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>example_counts2 <span class="ot">&lt;-</span> data_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"sample"</span>,<span class="at">value =</span> <span class="st">"gene_expression"</span>,<span class="dv">3</span><span class="sc">:</span><span class="dv">1219</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_name=</span> <span class="fu">fct_reorder</span>(gene_name,gene_expression ))</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(example_counts2, <span class="fu">aes</span>(<span class="at">x=</span>gene_name, <span class="at">y=</span>gene_expression, <span class="at">fill=</span>gene_name, <span class="at">color=</span>gene_name)) <span class="sc">+</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis</span>(<span class="at">discrete=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span> <span class="co"># Improve plot appearance</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span> ) <span class="sc">+</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="co"># flip the x and y axes</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Assigned Probability (%)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-2HorizontalViolin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-2HorizontalViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-2HorizontalViolin-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-2HorizontalViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Horizontal Violin Plot using the <code>TCGA-BRCA</code> dataset
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="violin-plot-with-boxplot" class="level3">
<h3 class="anchored" data-anchor-id="violin-plot-with-boxplot">3. Violin Plot with Boxplot</h3>
<p>In practical visualization applications, a box plot can be added to a violin plot using <code>geom_boxplot()</code>, which helps in visually comparing the distribution of the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>example_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_size) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">myaxis =</span> <span class="fu">paste0</span>(name, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"n="</span>, num)) <span class="co"># The `myaxis` variable is created to display sample size on the x-axis.  </span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(example_data, <span class="fu">aes</span>(<span class="at">x=</span>myaxis, <span class="at">y=</span>value, <span class="at">fill=</span>name)) <span class="sc">+</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">width=</span><span class="fl">1.4</span>) <span class="sc">+</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_boxplot</span>( <span class="at">width=</span><span class="fl">0.1</span>,<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">alpha=</span><span class="fl">0.2</span>) <span class="sc">+</span> <span class="co"># Draw a box plot. A small width value makes the box plot inside the violin plot.</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span> <span class="co"># Beautify the graph</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>   <span class="at">legend.position=</span><span class="st">"none"</span>,</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>   <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">11</span>)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"A Violin plot wrapping a boxplot"</span>) <span class="sc">+</span>  <span class="co"># Set the title</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-3.1Violinwizbox" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-3.1Violinwizbox-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-3.1Violinwizbox-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-3.1Violinwizbox-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Violin Plot with Boxplot
</figcaption>
</figure>
</div>
</div>
</div>
<p>Another Violin Plot with Box Plot using the TCGA-BRCA Gene Expression Data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>example_counts3 <span class="ot">&lt;-</span> data_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,] <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"sample"</span>, <span class="at">value =</span> <span class="st">"gene_expression"</span>,<span class="dv">3</span><span class="sc">:</span><span class="dv">1219</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_name=</span> <span class="fu">fct_reorder</span>(gene_name,gene_expression ))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(example_counts3, <span class="fu">aes</span>(<span class="at">x=</span>gene_name, <span class="at">y=</span>gene_expression, <span class="at">fill=</span>gene_name, <span class="at">color=</span>gene_name)) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>( <span class="at">width=</span><span class="fl">0.1</span>,<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">alpha=</span><span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis</span>(<span class="at">discrete=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span> <span class="co"># Beautify the graph</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span> ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-3.2Violinwizbox" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-3.2Violinwizbox-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-3.2Violinwizbox-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-3.2Violinwizbox-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Violin Plot with Boxplot using the <code>TCGA-BRCA</code> dataset
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="grouped-violin-plot" class="level3">
<h3 class="anchored" data-anchor-id="grouped-violin-plot">4. Grouped Violin Plot</h3>
<p>On the basis of the basic violin plot, we can achieve intra-group comparison by setting the fill value.</p>
<p>The example below demonstrates intra-group comparison using the <code>fill</code> aesthetic. In this case, the <code>penguins</code> dataset is used. The x variable represents the species, and <code>fill=sex</code> creates an intra-group classification to visualize the comparison of flipper lengths within each species, grouped by gender.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_penguins, <span class="fu">aes</span>(<span class="at">fill=</span>sex, <span class="at">y=</span>flipper_length_mm, <span class="at">x=</span>species)) <span class="sc">+</span> <span class="co"># Use X as the major classification and fill as the intra-group classification</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">position=</span><span class="st">"dodge"</span>, <span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">outlier.colour=</span><span class="st">"transparent"</span>) <span class="sc">+</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span>T, <span class="at">name=</span><span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>()  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-4GroupedViolin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-4GroupedViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-4GroupedViolin-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-4GroupedViolin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Grouped Violin Plot using the <code>penguins</code> dataset
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="half-violin-plot" class="level3">
<h3 class="anchored" data-anchor-id="half-violin-plot">5. Half-Violin Plot</h3>
<p>A half-violin plot is useful for visualizing a large amount of data in a compact form. We can use the <code>geom_half_violin</code> function to display two groups separately.</p>
<p>In the following example, we visualize the flipper lengths of penguins for both species and genders, by plotting female and male penguins on opposite sides of the plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate the data for female and male penguins</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>data_female <span class="ot">&lt;-</span> data_penguins <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sex <span class="sc">==</span> <span class="st">"female"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>data_male <span class="ot">&lt;-</span> data_penguins <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sex <span class="sc">==</span> <span class="st">"male"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the half-violin plot for both groups (females on the right and males on the left)</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_half_violin</span>(</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> data_female,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> flipper_length_mm, <span class="at">x =</span> species),</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="dv">1</span>),</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="st">'width'</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="cn">NA</span>,</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"#9370DB"</span>,</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.8</span>,  <span class="do">## Set transparency</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">side =</span> <span class="st">"r"</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_half_violin</span>(</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> data_male,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> flipper_length_mm, <span class="at">x =</span> species),</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="dv">1</span>),</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="st">'width'</span>,</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="cn">NA</span>,</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"#FFFF00"</span>,</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.6</span>,</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">side =</span> <span class="st">"l"</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-5Half-ViolinPlot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-5Half-ViolinPlot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-5Half-ViolinPlot-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-5Half-ViolinPlot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Half-Violin Plot using the <code>penguins</code> dataset
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="violin-plot-using-the-ggstatsplot-package" class="level3">
<h3 class="anchored" data-anchor-id="violin-plot-using-the-ggstatsplot-package">6. Violin Plot using the <code>ggstatsplot</code> Package</h3>
<p>The <code>ggstatsplot</code> package extends <code>ggplot2</code> by adding powerful statistical visualizations. The <code>ggbetweenstats()</code> function allows the creation of combined violin plots, box plots, and scatter plots.</p>
<p>In the following example, we visualize the distribution of bill lengths across different penguin species using the <code>penguins</code> dataset. We further enhance the plotâ€™s aesthetics using the <code>theme()</code> function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>plt <span class="ot">&lt;-</span> <span class="fu">ggbetweenstats</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_penguins,</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> species,</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> bill_length_mm</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Beautification</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(  <span class="do">## Add labels and title</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Penguins Species"</span>,</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Bill Length"</span>,</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Distribution of bill length across penguins species"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey50"</span>),</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"#b4aea9"</span>),</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">linetype =</span> <span class="st">"dashed"</span>),</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"#fbf9f4"</span>, <span class="at">color =</span> <span class="st">"#fbf9f4"</span>),</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"#fbf9f4"</span>, <span class="at">color =</span> <span class="st">"#fbf9f4"</span>)</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-6ViolinPlot-ggstatsplot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-6ViolinPlot-ggstatsplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ViolinPlot_files/figure-html/fig-6ViolinPlot-ggstatsplot-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-6ViolinPlot-ggstatsplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Violin Plot using the <code>ggstatsplot</code> Package
</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="application" class="level2">
<h2 class="anchored" data-anchor-id="application">Application</h2>
<section id="basic-violin-plot-1" class="level3">
<h3 class="anchored" data-anchor-id="basic-violin-plot-1">1. Basic Violin Plot</h3>
<div id="fig-ViolinPlotApplications" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" alt="ViolinPlotApp1">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ViolinPlotApplications-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/ViolinPlot_app1.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%" alt="ViolinPlotApp1">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ViolinPlotApplications-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Applications of Basic Violin Plot
</figcaption>
</figure>
</div>
<p><a href="#fig-ViolinPlotApplications" class="quarto-xref">Figure&nbsp;10</a> e is a violin plot of the relative abundances of circRNAs in seven types of cancer tissues and their corresponding normal tissues [1].</p>
</section>
<section id="grouped-violin-plot-1" class="level3">
<h3 class="anchored" data-anchor-id="grouped-violin-plot-1">2. Grouped Violin Plot</h3>
<div id="fig-ViolinPlotApplications2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" alt="ViolinPlotApp2">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ViolinPlotApplications2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/ViolinPlot_app2.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%" alt="ViolinPlotApp2">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ViolinPlotApplications2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Applications of Grouped Violin Plot
</figcaption>
</figure>
</div>
<p>The above violin plot analyzes and compares the levels and distributions of 31 proteins and mRNAs in single A549 cells [2].</p>
</section>
<section id="half-violin-plot-1" class="level3">
<h3 class="anchored" data-anchor-id="half-violin-plot-1">3. Half-Violin Plot</h3>
<div id="fig-ViolinPlotApplications3" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" alt="ViolinPlotApp3">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ViolinPlotApplications3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/ViolinPlot_app3.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%" alt="ViolinPlotApp3">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ViolinPlotApplications3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Applications of Half-Violin Plot
</figcaption>
</figure>
</div>
<p><a href="#fig-ViolinPlotApplications3" class="quarto-xref">Figure&nbsp;12</a> E uses a half-violin plot to analyze the clone size distribution of WT clones in WT intestines (left) or Mâˆ’/+ intestines (middle and right figures) [3].</p>
</section>
<section id="violin-plot-with-boxplot-1" class="level3">
<h3 class="anchored" data-anchor-id="violin-plot-with-boxplot-1">4. Violin Plot with Boxplot</h3>
<div id="fig-ViolinPlotApplications4" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" alt="ViolinPlotApp4">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ViolinPlotApplications4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/ViolinPlot_app4.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%" alt="ViolinPlotApp4">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ViolinPlotApplications4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Applications of Violin Plot with Boxplot
</figcaption>
</figure>
</div>
<p><a href="#fig-ViolinPlotApplications4" class="quarto-xref">Figure&nbsp;13</a> D shows the expected median MSD and distribution of droplets attached to the substrate (noise) and in water (stimulation), as well as the nanoscale RMS displacement [4].</p>
</section>
</section>
<section id="reference" class="level2">
<h2 class="anchored" data-anchor-id="reference">Reference</h2>
<p>[1] Zheng Q, Bao C, Guo W, Li S, Chen J, Chen B, Luo Y, Lyu D, Li Y, Shi G, Liang L, Gu J, He X, Huang S. Circular RNA profiling reveals an abundant circHIPK3 that regulates cell growth by sponging multiple miRNAs. Nat Commun. 2016 Apr 6;7:11215. doi: 10.1038/ncomms11215. PMID: 27050392; PMCID: PMC4823868.</p>
<p>[2] Gong H, Wang X, Liu B, Boutet S, Holcomb I, Dakshinamoorthy G, Ooi A, Sanada C, Sun G, Ramakrishnan R. Single-cell protein-mRNA correlation analysis enabled by multiplexed dual-analyte co-detection. Sci Rep.&nbsp;2017 Jun 5;7(1):2776. doi: 10.1038/s41598-017-03057-5. PMID: 28584233; PMCID: PMC5459813.</p>
<p>[3] Kolahgar G, Suijkerbuijk SJ, Kucinski I, Poirier EZ, Mansour S, Simons BD, Piddini E. Cell Competition Modifies Adult Stem Cell and Tissue Population Dynamics in a JAK-STAT-Dependent Manner. Dev Cell. 2015 Aug 10;34(3):297-309. doi: 10.1016/j.devcel.2015.06.010. Epub 2015 Jul 23. PMID: 26212135; PMCID: PMC4537514.</p>
<p>[4] Cribb JA, Osborne LD, Beicker K, Psioda M, Chen J, Oâ€™Brien ET, Taylor Ii RM, Vicci L, Hsiao JP, Shao C, Falvo M, Ibrahim JG, Wood KC, Blobe GC, Superfine R. An Automated High-throughput Array Microscope for Cancer Cell Mechanics. Sci Rep.&nbsp;2016 Jun 6;6:27371. doi: 10.1038/srep27371. PMID: 27265611; PMCID: PMC4893602.</p>
<p>[5] Wickham H, Vaughan D, Girlich M (2024). <em>tidyr: Tidy Messy Data</em>. R package version 1.3.1, <a href="https://CRAN.R-project.org/package=tidyr" class="uri">https://CRAN.R-project.org/package=tidyr</a>.</p>
<p>[6] H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2016.</p>
<p>[7] Wickham H, FranÃ§ois R, Henry L, MÃ¼ller K, Vaughan D (2023). <em>dplyr: A Grammar of Data Manipulation</em>. R package version 1.1.4, <a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>.</p>
<p>[8] Tiedemann F (2022). <em>gghalves: Compose Half-Half Plots Using Your Favourite Geoms</em>. R package version 0.1.4, <a href="https://CRAN.R-project.org/package=gghalves" class="uri">https://CRAN.R-project.org/package=gghalves</a>.</p>
<p>[9] Wickham H (2023). <em>forcats: Tools for Working with Categorical Variables (Factors)</em>. R package version 1.0.0, <a href="https://CRAN.R-project.org/package=forcats" class="uri">https://CRAN.R-project.org/package=forcats</a>.</p>
<p>[10] Rudis B (2024). <em>hrbrthemes: Additional Themes, Theme Components and Utilities for â€˜ggplot2â€™</em>. R package version 0.8.7, <a href="https://CRAN.R-project.org/package=hrbrthemes" class="uri">https://CRAN.R-project.org/package=hrbrthemes</a>.</p>
<p>[11] Simon Garnier, Noam Ross, Robert Rudis, AntÃ´nio P. Camargo, Marco Sciaini, and CÃ©dric Scherer (2024). viridis(Lite) - Colorblind-Friendly Color Maps for R. viridis package version 0.6.5.</p>
<p>[12] Patil, I. (2021). Visualizations with statistical details: The â€˜ggstatsplotâ€™ approach. Journal of Open Source Software, 6(61), 3167, doi:10.21105/joss.03167</p>
<p>[13] Horst AM, Hill AP, Gorman KB (2020). palmerpenguins: Palmer Archipelago (Antarctica) penguin data. R package version 0.1.0. https://allisonhorst.github.io/palmerpenguins/. doi: 10.5281/zenodo.3960218.</p>
</section>
<section id="contributors" class="level2">
<h2 class="anchored" data-anchor-id="contributors">Contributors</h2>
<ul>
<li><strong>Editor</strong>: Yinying Chai</li>
<li><strong>Reviewers</strong>: ShiXiang Wang, Peng Luo, Hong Yang, KeXin Li, Ying Shi</li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("http:\/\/genaimed\.org\/Bizard\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="DrRobinLuo/Bizard" data-repo-id="R_kgDONWVY7Q" data-category="Announcements" data-category-id="DIC_kwDONWVY7c4ClcJT" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="preferred_color_scheme" data-lang="en" crossorigin="anonymous" data-loading="lazy" async="">
</script>
<input type="hidden" id="giscus-base-theme" value="preferred_color_scheme">
<input type="hidden" id="giscus-alt-theme" value="preferred_color_scheme">
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../Distribution/Histogram.html" class="pagination-link" aria-label="Histogram">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Histogram</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../About.html" class="pagination-link" aria-label="About Us">
        <span class="nav-page-text">About Us</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Â© CC-By Peng Lab, 2024-2025. All rights reserved.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://clustrmaps.com/site/1c4xe" title="Visit tracker"><img src="//clustrmaps.com/map_v2.png?cl=ffffff&amp;w=a&amp;t=n&amp;d=VK6_Uhjas4vA0CDps3EFeB0Fotb8hU50SYT4Fcq5nUI&amp;co=91c5eb&amp;ct=000000"></a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/DrRobinLuo/Bizard/blob/main/Distribution/ViolinPlot.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/DrRobinLuo/Bizard/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>